src/MatchResult/AbstractMatchResult.php

```diff
@@ -11,6 +11,9 @@
 
 abstract class AbstractMatchResult implements MatchResultInterface, JsonSerializable
 {
+    public const ID = null;
```

Comment:
Let's add types for these constants

Reply:
It's only in php 8.3 :(
Also the idea here is the constants should be strings in inherited classes - the `getId`/`getName` methods won't work until constants are explicitly set.